### PR TITLE
Check for non-zero length before calling list2env()

### LINKED ISSUE #1444: Peculiar environment Issue in ggplot2_2.0.0: "Error in list2env(members, envir = e) " running R 3.1.1
Error in ggplot2 running R 3.1.1:
```
install.packages("ggplot2")
library(ggplot2)

a<-1:3
b<-4:6
c <- as.data.frame(cbind(a,b))

qplot(a,b,data=c)
```

This yields the following error:

```
Error in list2env(members, envir = e) : 
  names(x) must be a character vector of the same length as x
```

Please note that I'm running R on a Windows Machine with mapped network drives, as such UNC paths are not supported in R and I made appropriate changes in the .Renviron file using R_LIBS.

Session Info as below:

```
sessionInfo()
R version 3.1.1 (2014-07-10)
Platform: i386-w64-mingw32/i386 (32-bit)

locale:
[1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252    
LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                       
LC_TIME=English_Australia.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_2.0.0

loaded via a namespace (and not attached):
[1] colorspace_1.2-6 grid_3.1.1       gtable_0.1.2     munsell_0.4.2
    plyr_1.8.3       Rcpp_0.12.2      scales_0.3.0     tools_3.1.1    
```

### PR DESCRIPTION
**Pull request by wch**
This fixes #1444. I'd suggest making a release after merging, since this a blocker for many people.

### CODE REVIEW

**Comment by wch** [comment]
For future reference, the issue was fixed in R 3.1.2.
https://github.com/wch/r-source/commit/e9609339a1772c4b5cb42e093c98142b06929f5c

File: NEWS.md [hunk 1]
```diff
@@ -1,7 +1,10 @@
-# ggplot2 2.0.0.9000
+# ggplot2 2.0.0.9xxx
```
Comment (hadley) [inline NEWS.md]: It doesn't really matter, but I prefer to just leave it as `.9000`

Comment (wch) [inline NEWS.md]: OK, I'll change it back.

**Comment by hadley** [comment]
LGTM. Want to squash it when you merge?

Actually can you also bump the required version to 3.1?

**Comment by wch** [comment]
OK, I bumped the R version and rebased into one commit.

**Comment by jonrobinson2** [comment]
Thank you! Any idea when this will be pushed to CRAN for release?

**Comment by jonrobinson2** [comment]
Is this being made into a release? Would love to be able to bump our version of ggplot2.

**Comment by steinmb** [comment]
> Is this being made into a release?

No. I had to install from HEAD. It broke ggplot on raspberryPI (Debian Jessie) that installs R 3.1.x. This is how I worked around it:

Optional step. Only needed If you do not have devtools already installed in R.

    install.packages("devtools")

Install development version of ggplot2

    devtools::install_github("hadley/ggplot2")

**Pull request merged**